app/controllers/digital-wallet/post-google-pay-controller.js

```diff
@@ -11,14 +12,36 @@ module.exports = async (req, res) => {
   const gatewayAccountId = req.account.gateway_account_id
   const correlationId = req.headers[CORRELATION_HEADER] || ''
   const enable = req.body['google-pay'] === 'on'
+  const gatewayMerchantId = req.body.merchantId
+
+  if (enable && !gatewayMerchantId) {
+    req.flash('genericError', '<h2>Please enter a valid Merchant ID</h2>')
+    return res.redirect(paths.digitalWallet.googlePay)
+  }
+
+  if (enable) {
```

Comment:
I know this is following the previous code convention (which is a good thing!) but the logic here is quite hard to follow and might be simpler if it just did error handling once? 

Something like: 
```js
try {
  if (enablingGooglePay) {
    await connector.setGatewayMerchantId(gatewayAccountId, gatewayMerchantId, correlationId)
  }
  await connector.toggleGooglePay(gatewayAccountId, enable, correlationId)
  logger.info('...')
  req.flash('generic', '...')
} catch (error) {
  logger.error('...')
  req.flash('error', '...')
}
```

Reply:
The reason why I did it separately is because we rely on connector for the validation, so wanted to be sure that we're only showing a validation error when appropriate